otel-agent/ecs-ec2/config.yaml

```diff
@@ -1,74 +1,99 @@
 receivers:
-  hostmetrics:
-    collection_interval: 60s
-    root_path: /hostfs
-    scrapers:
-      load:
-      filesystem:
-      memory:
-      network:
-      paging:
-      cpu:
-      disk:
-      process:
-        exclude:
-          names: [ otelcol-contrib ]
-          match_type: strict
-        mute_process_name_error: true
-  prometheus:
-    config:
-      scrape_configs:
-        - job_name: 'otel-collector'
-          scrape_interval: 60s
-          static_configs:
-            - targets: ['0.0.0.0:8888']
   otlp:
     protocols:
       grpc:
-        endpoint: 0.0.0.0:4317
+        endpoint: "0.0.0.0:4317"
       http:
-        endpoint: 0.0.0.0:4318
+        endpoint: "0.0.0.0:4318"
+
+  filelog:
+    start_at: end
+    include:
+      - /hostfs/var/lib/docker/containers/*/*.log
+      
+    include_file_path: true
+
+    # add log.file.name to resource attributes
+    operators:
+      - type: move
+        from: attributes["log.file.name"]
+        to: resource["log.file.name"]
+
+  # add metrics
+  awsecscontainermetricsd:
+
 processors:
-  batch:
-  batch/metrics:
-    timeout: 60s
   resourcedetection:
-    detectors: [env, ec2, ecs, docker, system]
+    # ecs & docker detectors not required when using ecslogresourcedetection for logs
+    detectors: [env, ec2, system]
     timeout: 2s
     override: false
+
+  ecsattributes:
+    container_id:
+      sources:
+        - "log.file.name"
+
+  # only allow logs with container.name label
+  filter/ecslogs:
+    logs:
+      include:
+        match_type: regexp
+        resource_attributes:
+          - key: aws.ecs.container.name
+            value: '.*'
+
+  transform/logs:
+    error_mode: ignore
+    log_statements:
+      - context: resource
+        statements:
+          - set(attributes["cx_container_id"], attributes["docker.id"])
+          - set(attributes["aws_ecs_task_family"], attributes["aws.ecs.task.definition.family"])
+          - set(attributes["image_id"], attributes["image.id"])
+          - delete_key(attributes, "image.id")
```

Comment:
Is there reason for different configurations for `transform/logs` and `ecsattributes` between the config and logging YAML?

Reply:
There's no reason other than to show different ways of configuring the processors. The configuration above specifically renames the image.id attribute to image_id to prevent mapping exception within the platform